### PR TITLE
move state-munging utils function into their own module

### PR DESCRIPTION
**Pull request by neilvyas**

### CODE REVIEW

**Comment by neilvyas** [comment]
lol deleted the wrong branch upstream, 🤷‍♂️

File: src/chipper/ui.cljs [hunk 1]
```diff
@@ -25,7 +25,7 @@
     [:span {:id :save
             :class (str "button"
                         (when (:frame-edited @state) " bright-text"))
-            :on-click #(utils/save! state)}
+            :on-click #(s/save! state)}  ; TODO FIXME don't call save-state directly, make it an action.
```
Comment (cqsd) [inline src/chipper/ui.cljs]: yeah,

File: src/chipper/state.cljs [hunk 2]
```diff
@@ -0,0 +1,148 @@
+(ns chipper.state
+  (:require [clojure.string :refer [split-lines]]
+            [chipper.notes :as notes]))  ; not good to require this in utils
+
+(defn reset-cursor! [state]
```
Comment (cqsd) [inline src/chipper/state.cljs]: hm, i should probably put all the state-changing stuff from keyboard into here too

**Comment by cqsd** [comment]
lgtm, good change, merge whenever

File: src/chipper/state.cljs (hunk 2 above)
Comment (cqsd) [reply src/chipper/state.cljs]: Or, really, there should be an `actions`

File: src/chipper/state.cljs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  one frame per line
+  each slice is a sequence of four notes
+  each note is a sequence of 3 characters
+   - note is 0-B (0-11) inclusive; o for :off, s for :stop
+   - octave is 0-9 inclusiv
+   - gain is 0-9 inclusive"
+  [frames]
+  (apply str (interpose "\n" (map serialize-frame frames))))
+
+(def note-reverse-map (zipmap (vals notes/note-steps) (keys notes/note-steps)))
```
Comment (cqsd) [inline src/chipper/state.cljs]: this is a ridiculously stupid function, and won't be necessary soon!

**Pull request merged**